src/throttler.guard.ts

```diff
@@ -106,6 +138,8 @@ export class ThrottlerGuard implements CanActivate {
     context: ExecutionContext,
     limit: number,
     ttl: number,
+    getTracker: ThrottlerGetTrackerFunction,
+    generateKey: ThrottlerGenerateKeyFunction,
```

Comment:
As this method is `protected` and can be overriden by others, can we change these to be at the end of the parameters list instead of in the middle? It'll ensure that those who _have_ overriden this don't get a breaking change here.

Reply:
I adjusted the method signature according to your suggestion, but if I'm not mistaken that would still be breaking manual overrides since these two parameters aren't optional but rather mandatory.
Making them optional would then again require to check for their existence status in the `handleRequest` method where we'd then possibly have to apply the default function from `this.commonOptions.<getTracker/generateKey>` instead of doing so in `canActivate`.

Reply:
By moving them to the end of the signature, the signature for the `handleRequest` method becomes `handleRequest(context: ExecutionContext, limit: number, ttl: number, throttler: ThrottlerOptions, getTracker: ThrottlerGetTrackerFunction, generateKey: ThrottlerGenerateKeyFunction): boolean`. However, JS doesn't care if you don't provide all of the parameters, and TS is strangely okay with omitting parameters from the extended class as well, they will still end up being passed, because of how JS works, but they won't be referenced, so it's fine